### PR TITLE
Implement QUIT, JOIN, PART, TOPIC, NAMES, LIST, INVITE, KICK, PRIVMSG, NOTICE

### PR DESCRIPTION
**Pull request by JonathanDUFOUR**
Related to #12.
The IRC commands listed in the title have been implemented and tested manually only.
Also, the `RFCs_summary.html` file has been updated on some points to better explain the implementation details.

### CODE REVIEW

File: src/to_correct
Comment (Saphirac) [inline src/to_correct:1]: Please delete that

**Review by Saphirac — COMMENTED** [review]
Almost nothing to do, let's push that !

**Review by Saphirac — DISMISSED** [review]
go

**Review by JonathanDUFOUR — COMMENTED** [review]
I changed the last things that I estimated as incorrect or reasonably improvable.

**Review by Saphirac — APPROVED** [review]
Finished!

**Pull request merged**